### DISCUSSION TITLE
deploy to Vercel error(Build error occurred)

### DISCUSSION THREAD

**Discussion by 2ynhub** [vercel/next.js#14416] [Help] [answered]
Hi, I am new to GitHub, Next.js and Vercel. I am following https://nextjs.org/learn/basics/deploying-nextjs-app/deploy. I had a problem with deployment.

This is my github https://github.com/2ynhub/nextjs-blog. Below is the log of build at Vercel.
What is the problem here?

```
01:18:57.307  Cloning github.com/2ynhub/nextjs-blog (Branch: master, Commit: 850dd65)
01:18:58.072  Cloning completed in 764ms
01:18:58.072  Analyzing source code...
01:18:59.782  Installing build runtime...
01:19:00.194  Build runtime installed: 411.956ms
01:19:00.613  Looking up build cache...
01:19:00.654  Build cache not found
01:19:01.080  Installing dependencies...
01:19:11.541  npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
01:19:11.541  npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
01:19:11.541  npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
01:19:11.541  npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
01:19:11.544  added 859 packages from 346 contributors in 10.061s
01:19:11.902  79 packages are looking for funding
01:19:11.902    run `npm fund` for details
01:19:11.953  Running "npm run build"
01:19:12.143  > learn-starter@0.1.0 build /vercel/4df112c7
01:19:12.143  > next build
01:19:12.601  Browserslist: caniuse-lite is outdated. Please run next command `npm update`
01:19:12.670  Warning: No build cache found. Please configure build caching for faster rebuilds. Read more: https://err.sh/next.js/no-cache
01:19:12.671  Creating an optimized production build...
01:19:12.685  Attention: Next.js now collects completely anonymous telemetry regarding usage.
01:19:12.685  This information is used to shape Next.js' roadmap and prioritize features.
01:19:12.685  You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
01:19:12.685  https://nextjs.org/telemetry
01:19:18.757  Failed to compile.
01:19:18.757  ./components/date.js
01:19:18.757  Module not found: Can't resolve 'date-fns' in '/vercel/4df112c7/components'
01:19:18.758  > Build error occurred
01:19:18.758  Error: > Build failed because of webpack errors
01:19:18.758      at build (/vercel/4df112c7/node_modules/next/dist/build/index.js:13:900)
01:19:18.769  npm ERR! code ELIFECYCLE
01:19:18.769  npm ERR! errno 1
01:19:18.770  npm ERR! learn-starter@0.1.0 build: `next build`
01:19:18.770  npm ERR! Exit status 1
01:19:18.770  npm ERR! 
01:19:18.770  npm ERR! Failed at the learn-starter@0.1.0 build script.
01:19:18.770  npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
01:19:18.774  npm ERR! A complete log of this run can be found in:
01:19:18.774  npm ERR!     /vercel/.npm/_logs/2020-06-21T16_19_18_771Z-debug.log
01:19:18.778  Error: Command "npm run build" exited with 1
01:19:20.299  Done with "package.json"
```

**Comment by jamesmosier** [comment, marked as answer]
You are missing `date-fns` from your `package.json` and it is used within `components/date.js`

Try adding it then redeploying.
```bash
npm install date-fns --save
```

**Comment by hari561** [comment]
Hi 

I have a problem in deploying the Next.js application through vercel. below is the log file. Can you please help me?


08:56:52 | Cloning github.com/hari561/nextjs-blog (Branch: main, Commit: 8dfadb3)
-- | --
08:56:52 | Cloning completed in 428ms
08:56:52 | Analyzing source code...
08:56:53 | Installing build runtime...
08:56:55 | Build runtime installed: 2627.459ms
08:56:58 | Looking up build cache...
08:56:58 | Build cache not found
08:56:59 | Installing dependencies...
08:57:07 | > sharp@0.26.3 install /vercel/workpath0/node_modules/sharp
08:57:07 | > (node install/libvips && node install/dll-copy && prebuild-install) \|\| (node-gyp rebuild && node install/dll-copy)
08:57:07 | info sharp Using cached /vercel/.npm/_libvips/libvips-8.10.0-linux-x64.tar.br
08:57:09 | > @ampproject/toolbox-optimizer@2.7.1-alpha.0 postinstall /vercel/workpath0/node_modules/@ampproject/toolbox-optimizer
08:57:09 | > node lib/warmup.js
08:57:09 | Browserslist: caniuse-lite is outdated. Please run:
08:57:09 | npx browserslist@latest --update-db
08:57:09 | AMP OPTIMIZER Downloaded latest AMP runtime data.
08:57:10 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.1 (node_modules/fsevents):
08:57:10 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
08:57:10 | added 377 packages from 413 contributors in 10.157s
08:57:10 | 72 packages are looking for funding
08:57:10 | run `npm fund` for details
08:57:10 | Running "npm run build"
08:57:10 | > learn-starter@0.1.0 build /vercel/workpath0
08:57:10 | > next build
08:57:11 | info  - Creating an optimized production build...
08:57:11 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
08:57:11 | This information is used to shape Next.js' roadmap and prioritize features.
08:57:11 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
08:57:11 | https://nextjs.org/telemetry
08:57:12 | Failed to compile.
08:57:12 | ModuleNotFoundError: Module not found: Error: Can't resolve './layout.module.css' in '/vercel/workpath0/pages/posts'
08:57:12 | > Build error occurred
08:57:13 | Error: > Build failed because of webpack errors
08:57:13 | at /vercel/workpath0/node_modules/next/dist/build/index.js:15:918
08:57:13 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
08:57:13 | at async /vercel/workpath0/node_modules/next/dist/build/tracer.js:1:1441
08:57:13 | npm ERR! code ELIFECYCLE
08:57:13 | npm ERR! errno 1
08:57:13 | npm ERR! learn-starter@0.1.0 build: `next build`
08:57:13 | npm ERR! Exit status 1
08:57:13 | npm ERR!
08:57:13 | npm ERR! Failed at the learn-starter@0.1.0 build script.
08:57:13 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
08:57:13 | npm ERR! A complete log of this run can be found in:
08:57:13 | npm ERR!     /vercel/.npm/_logs/2021-01-31T13_57_13_020Z-debug.log
08:57:13 | Error: Command "npm run build" exited with 1
08:57:17 | Done with "package.json"

**Reply by jamesmosier** [reply]
You path is incorrect to your css file:

> `Can't resolve './layout.module.css' in '/vercel/workpath0/pages/posts'`

That means inside `pages/posts.js` you need to fix the path to `layout.module.css` so it can find that file.
